### PR TITLE
Propose an ibis blog post about new release 1.3

### PR DESCRIPTION
**Pull request by xmnlab**

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @xmnlab,  and sorry for the slow reply.

@datapythonista would you be able to help review this? 
There's a rendered version on CircleCI at https://321-179573943-gh.circle-artifacts.com/0/output/blog/2020/03/ibis-1.3-notes/index.html

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 1]
```diff
@@ -0,0 +1,1297 @@
...
+    "\n",
+    "Some hightlighted features of this new version are:\n",
+    "\n",
+    "* Renaming `mapd` backend to `omniscidb`\n",
+    "* Added new backends and geospatial methods to the documentation\n",
+    "* Improvement of geospatial support\n",
+    "* Addition of `PySpark` backend\n",
+    "* Addition of `JSON`, `JSONB` and `UUID` data types\n",
+    "* Support for `Python 3.5` dropped.\n",
+    "* Initial support for `Python 3.8` added.\n",
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:19]: The order here feels a bit weird to me. I'd expect this list to be sorted by importance, and I think renaming the omnisci backend is way less important than adding the PySpark backend, or adding Python 3.8 support.

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 2]
```diff
@@ -0,0 +1,1297 @@
...
+    "\n",
+    "This blog post is divided into different notes section, such as:\n",
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:21]: May be?
```suggestion
    "This blog post is divided into different sections:\n",
```

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 3]
```diff
@@ -0,0 +1,1297 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### OmniSciDB notes\n",
+    "\n",
+    "Remember, `mapd` backend now is `omniscidb`! \n",
+    "\n",
+    "Another important feature is that now you can define if the connection is `IPC` and you can also specify the `GPU` device ID you want to use it (if you have a NVIDIA card, supported by `cudf`)."
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:47]: I don't think the rename is a feature, so I wouldn't call this "another feature"

Also, may be explain briefly what's IPC?

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 4]
```diff
@@ -0,0 +1,1297 @@
...
+    "expr = t[['id', 'bool_col']].head(5)\n",
+    "df = expr.execute(ipc=False, gpu_device=None)\n",
+    "df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can imagine, the result of `df` is a `Pandas.DataFrame` "
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:171]:
```suggestion
    "As you can imagine, the result of `df` is a `pandas.DataFrame` "
```

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 5]
```diff
@@ -0,0 +1,1297 @@
...
+    "            pt.StructField('smallint_col', pt.ShortType(), True),\n",
+    "            pt.StructField('int_col', pt.IntegerType(), True),\n",
+    "            pt.StructField('bigint_col', pt.LongType(), True),\n",
+    "            pt.StructField('float_col', pt.FloatType(), True),\n",
+    "            pt.StructField('double_col', pt.DoubleType(), True),\n",
+    "            pt.StructField('date_string_col', pt.StringType(), True),\n",
+    "            pt.StructField('string_col', pt.StringType(), True),\n",
+    "            pt.StructField('timestamp_col', pt.TimestampType(), True),\n",
+    "            pt.StructField('year', pt.IntegerType(), True),\n",
+    "            pt.StructField('month', pt.IntegerType(), True),\n",
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:398]: Personally I think this example is to complex to illustrate a backend.

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 6]
```diff
@@ -0,0 +1,1297 @@
...
+   "source": [
+    "t = con_pyspark.table('functional_alltypes')\n",
+    "t"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Different than a `SQL` backend, that return a `SQL` statement, the returned \n",
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:455]:
```suggestion
    "Different than a `SQL` backend, that returns a `SQL` statement, the returned \n",
```

File: posts/2020/03/ibis-1.3-notes.ipynb [hunk 7]
```diff
@@ -0,0 +1,1297 @@
...
+    "    t_geo.geo_multipolygon, \n",
+    "    t_geo['geo_multipolygon'].contains(shp_point).name('contains_point_1')\n",
+    "].execute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So, as expected, just the first two `multipolygons` contains a `point` with coordinates `(20, 10)`.\n",
```
Comment (datapythonista) [inline posts/2020/03/ibis-1.3-notes.ipynb:1219]:
```suggestion
    "So, as expected, just the first two `multipolygons` contain a `point` with coordinates `(20, 10)`.\n",
```

**Comment by xmnlab** [comment]
@datapythonista thanks for the feedback .. I will work on the suggestions you did.

related to https://github.com/Quansight-Labs/quansight-labs-site/pull/64#discussion_r404086518

I am not sure yet how to improve this ... IIRC I did this because create_table didn't work well ... but I can try again .. maybe I missed something. thanks!

**Comment by xmnlab** [comment]
@datapythonista not sure if I could do much about the pyspark note comment, as I think the "setup" is important there as it is different than other backends. I tried to keep it more readable ... not sure if it is enough ... any suggestion would be very appreciated.

the new rendered version is here https://333-179573943-gh.circle-artifacts.com/0/output/blog/2020/04/ibis-1.3-notes/index.html

thanks!

**Comment by rgommers** [comment]
@datapythonista could you re-review this?

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 8]
```diff
@@ -0,0 +1,1366 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ibis framework 1.3 was just released, after 8 months of development work with 104 new commits from 16 distinct contributors. What is new? In this blog post we will check some important highlighted features about this new version!\n",
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:7]:
```suggestion
    "Ibis framework 1.3 was just released, after 8 months of development work, with 104 new commits from 16 unique contributors. What is new? In this blog post we will discuss some important features about this new version!\n",
```

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 9]
```diff
@@ -0,0 +1,1366 @@
...
+    "\n",
+    "First, if you are new to Ibis framework world, you can check this [blog-post](https://labs.quansight.org/blog/2019/07/ibis-python-data-analysis-productivity-framework/) wrote last year with some introductory information about that.\n",
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:9]:
```suggestion
    "First, if you are new to the Ibis framework world, you can check this [blog post](https://labs.quansight.org/blog/2019/07/ibis-python-data-analysis-productivity-framework/), wrote last year, with some introductory information about it.\n",
```

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 10]
```diff
@@ -0,0 +1,1366 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### OmniSciDB notes\n",
+    "\n",
+    "Remember, `mapd` backend now is `omniscidb`! \n",
+    "\n",
+    "A important feature on `omniscidb` is that now you can define if the connection is `IPC` and you can also specify the `GPU` device ID you want to use it (if you have a NVIDIA card, supported by `cudf`).\n",
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:49]:
```suggestion
    "An important feature of `omniscidb` is that now you can define if the connection is `IPC`, and you can also specify the `GPU` device ID you want to use (if you have a NVIDIA card, supported by `cudf`).\n",
```

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 11]
```diff
@@ -0,0 +1,1366 @@
...
+    "\n",
+    "The `IPC` or `Inter-Process Communication` is used to provide shared data support between process. OmniSciDB uses Apache Arrow to provide IPC support."
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:51]:
```suggestion
    "The `IPC` or `Inter-Process Communication` is used to provide shared data support between processes. OmniSciDB uses Apache Arrow to provide IPC support."
```

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 12]
```diff
@@ -0,0 +1,1366 @@
...
+    "    gpu_device=None\n",
+    ")\n",
+    "con_omni.list_tables()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Also now you can specify `ipc` or `gpu_device` directly to the `execute` method:"
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:87]:
```suggestion
    "Also you can now specify `ipc` or `gpu_device` directly to the `execute` method:"
```

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 13]
```diff
@@ -0,0 +1,1366 @@
...
+   "source": [
+    "jsonb_value = ibis.literal(json.dumps({\"id\": 1}).encode('utf8'), type='jsonb')\n",
+    "jsonb_value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Another important new features on `PostgreSQL` backend was some new `geospatial` operations support, such as\n",
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:318]:
```suggestion
    "Another important new features on `PostgreSQL` backend is the support of new `geospatial` operations, such as\n",
```

File: posts/2020/04/ibis-1.3-notes.ipynb [hunk 14]
```diff
@@ -0,0 +1,1366 @@
...
+    "* GeometryType\n",
+    "* GeometryN\n",
+    "* IsValid\n",
+    "* LineLocatePoint\n",
+    "* LineMerge\n",
+    "* LineSubstring\n",
+    "* OrderingEquals\n",
+    "* Union\n",
+    "\n",
+    "Also, it has now more two geospatial data type support: `MULTIPOINT` and `MULTILINESTRING`.\n",
```
Comment (datapythonista) [inline posts/2020/04/ibis-1.3-notes.ipynb:329]:
```suggestion
    "Also, now it has support for two geospatial data types: `MULTIPOINT` and `MULTILINESTRING`.\n",
```

**Comment by rgommers** [comment]
Great, thanks @datapythonista. Looking forward to hitting the green button here:)

**Comment by xmnlab** [comment]
thank you a lot @datapythonista ! <3

I applied all changes you suggested.

probably I will need to change the nikola-date for this blog post. what date would be convenient for this?

**Comment by rgommers** [comment]
Thanks @xmnlab! Don't worry about the date, I'll change it when I publish it. Which is not today, given that we already have a blog post from Carlos for today.

**Comment by xmnlab** [comment]
sounds good! thanks @rgommers and @datapythonista !

**Comment by rgommers** [comment]
Did some copy-editing, looks good to go now, so merged. Thanks @xmnlab, and thanks @datapythonista for the review.

**Pull request merged**